test/src/error-snippets/index.ts

```diff
@@ -0,0 +1,131 @@
+
+describe("error-snippets", () => {
+    const cloneError = (err: Error): Error => {
+        const newError = err;
+
+        newError.name = err.name;
+        newError.message = err.message;
+        newError.stack = err.stack;
+
+        return newError;
+    };
+
+    describe("extendWithCodeSnippet", () => {
+        const sandbox = sinon.createSandbox();
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        let extendWithCodeSnippet: (err?: any) => Promise<Error & { snippet?: string }>;
+        let fsReadFileStub: SinonStub;
+        let fetchStub: SinonStub;
+        let formatErrorSnippetStub: SinonStub;
+        let findRelevantStackFrameStub: SinonStub;
+        let loggerStub: { warn: SinonStub };
+
+        beforeEach(() => {
+            fsReadFileStub = sandbox.stub(fs, "readFile");
+            fetchStub = sandbox.stub(globalThis, "fetch");
+
+            formatErrorSnippetStub = sandbox.stub();
+            findRelevantStackFrameStub = sandbox.stub();
+            loggerStub = { warn: sandbox.stub() };
+
+            extendWithCodeSnippet = proxyquire("../../../src/error-snippets", {
+                "./utils": { formatErrorSnippet: formatErrorSnippetStub },
+                "./frames": { findRelevantStackFrame: findRelevantStackFrameStub },
+                "../utils/logger": loggerStub,
+            }).extendWithCodeSnippet;
+        });
+
+        afterEach(() => sandbox.restore());
+
+        it("should not modify error, if it is falsy", async () => {
+            await assert.eventually.isUndefined(extendWithCodeSnippet());
+            await assert.eventually.isNull(extendWithCodeSnippet(null));
+            await assert.eventually.equal(extendWithCodeSnippet(""), "");
+
+            assert.notCalled(loggerStub.warn);
+        });
+
+        it("should not modify error, if could not find relevant stack frame", async () => {
+            const error = new Error("test error");
+            const savedError = cloneError(error);
+            findRelevantStackFrameStub.returns(null);
+
+            await extendWithCodeSnippet(error);
+
+            assert.notCalled(loggerStub.warn);
+            assert.strictEqual(error, savedError);
+        });
+
+        it("should not modify error, if stack frame resolver fails", async () => {
+            const error = new Error("test error");
+            const savedError = cloneError(error);
+            const stackFrame = { fileName: "/foo/file1" };
+            findRelevantStackFrameStub.returns(stackFrame);
+            fsReadFileStub.withArgs("/foo/file1").rejects(new Error("read file error"));
+
+            const result = await extendWithCodeSnippet(error);
+
+            assert.strictEqual(error, savedError);
+            assert.strictEqual(error, result);
+            assert.calledOnceWith(loggerStub.warn, "Unable to apply code snippet:");
+        });
+
+        it("should return the same Object.is object", async () => {
```

Comment:
not sure that the name of the test needs to be specified about `Object.is`

Reply:
Why not?
`{}` and `{}` are the same objects. But there is a nuance)